lib/lpc55-puf/src/lib.rs

```diff
@@ -6,8 +6,20 @@
 
 use core::mem;
 use lpc55_pac::PUF;
+use num_derive::FromPrimitive;
+use num_traits::FromPrimitive;
 use unwrap_lite::UnwrapLite;
 
+/// Used to represent valid states for PUF index blocking bits in IDXBLK_L &
+/// IDXBLK_H registers. We derive FromPrimivite for this type to enable use
+/// of 'from_u32'. This will map invalid / reserved register states to `None`.
+#[repr(u32)]
```

Comment:
(tip: you don't need `repr(u32)` here, that only matters if you're going to be casting pointers to it or otherwise doing bit manipulation. `FromPrimitive` removes the need for that.)